File: BackendAPI/VitroFit.API/Entities/Workout.cs
using System.ComponentModel.DataAnnotations;

namespace VitroFit.API.Entities
{
    /// <summary>Shared catalog of workouts users can schedule into their timetable.</summary>
    public class Workout
    {
        public int Id { get; set; }

        [Required]
        [MaxLength(100)]
        public string Name { get; set; } = string.Empty;

        [Required]
        [MaxLength(100)]
        public string Category { get; set; } = string.Empty;

        [MaxLength(500)]
        public string? Description { get; set; }

        public ICollection<TimetableSlot> TimetableSlots { get; set; } = new List<TimetableSlot>();
    }
}

File: vitrofit_mobile/lib/models/workout.dart
class Workout {
  final int id;
  final String name;
  final String category;
  final String? description;

  const Workout({
    required this.id,
    required this.name,
    required this.category,
    this.description,
  });

  factory Workout.fromJson(Map<String, dynamic> json) {
    return Workout(
      id: json['id'] as int,
      name: (json['name'] as String?) ?? '',
      category: (json['category'] as String?) ?? '',
      description: json['description'] as String?,
    );
  }
}
